File: python/tests/test_unit_converter.py
"""
Tests for Unit Converter module
"""

import pytest
from cps_utility_console.functions.unit_converter import convert_temperature, convert_pressure


def test_temperature_c_to_f():
    """Test Celsius to Fahrenheit."""
    assert convert_temperature(0, "C", "F") == 32.0
    assert convert_temperature(100, "C", "F") == 212.0


def test_temperature_k_to_c():
    """Test Kelvin to Celsius."""
    assert convert_temperature(273.15, "K", "C") == 0.0


def test_temperature_invalid_unit():
    """Test invalid unit raises error."""
    with pytest.raises(ValueError, match="Units must be one of"):
        convert_temperature(100, "C", "X")


def test_pressure_bar_to_kpa():
    """Test bar to kPa."""
    assert convert_pressure(1, "BAR", "KPA") == 100.0


def test_pressure_negative():
    """Test negative pressure raises error."""
    with pytest.raises(ValueError, match="cannot be negative"):
        convert_pressure(-5, "BAR", "KPA")

File: cpp/tests/test_unit_converter.cpp
#include <gtest/gtest.h>
#include "unit_converter.hpp"

TEST(UnitConverter, CelsiusToFahrenheit) {
    EXPECT_NEAR(cps::convertTemperature(0.0f, "C", "F"), 32.0f, 0.001f);
    EXPECT_NEAR(cps::convertTemperature(100.0f, "C", "F"), 212.0f, 0.001f);
}

TEST(UnitConverter, CelsiusToKelvin) {
    EXPECT_NEAR(cps::convertTemperature(0.0f, "C", "K"), 273.15f, 0.001f);
}

TEST(UnitConverter, InvalidUnitThrows) {
    EXPECT_THROW(cps::convertTemperature(0.0f, "X", "C"), std::invalid_argument);
}

TEST(UnitConverter, BelowAbsoluteZeroThrows) {
    EXPECT_THROW(cps::convertTemperature(-300.0f, "C", "K"), std::invalid_argument);
}

TEST(UnitConverter, BarToKpa) {
    EXPECT_NEAR(cps::convertPressure(1.0f, "BAR", "KPA"), 100.0f, 0.001f);
}

TEST(UnitConverter, NegativePressureThrows) {
    EXPECT_THROW(cps::convertPressure(-1.0f, "BAR", "KPA"), std::invalid_argument);
}